druid/src/app.rs

```diff
@@ -37,27 +35,75 @@ pub struct AppLauncher<T> {
     ext_event_host: ExtEventHost,
 }
 
-/// A description of a window to be instantiated.
-///
-/// This includes a function that can build the root widget, as well as other
-/// window properties such as the title.
-pub struct WindowDesc<T> {
-    pub(crate) root: Box<dyn Widget<T>>,
-    pub(crate) title: LabelText<T>,
+/// Window configuration that can be applied to a WindowBuilder, or to an existing WindowHandle.
+/// It does not include anything related to app data.
+pub struct WindowConfig {
     pub(crate) size: Option<Size>,
     pub(crate) min_size: Option<Size>,
     pub(crate) position: Option<Point>,
-    pub(crate) menu: Option<MenuDesc<T>>,
-    pub(crate) resizable: bool,
-    pub(crate) show_titlebar: bool,
-    pub(crate) state: WindowState,
+    pub(crate) resizable: Option<bool>,
+    pub(crate) show_titlebar: Option<bool>,
+    pub(crate) level: Option<WindowLevel>,
+    pub(crate) state: Option<WindowState>,
+}
+
+/// A description of a window to be instantiated.
+pub struct WindowDesc<T> {
+    pub(crate) pending: PendingWindow<T>,
+    pub(crate) config: WindowConfig,
     /// The `WindowId` that will be assigned to this window.
     ///
     /// This can be used to track a window from when it is launched and when
     /// it actually connects.
     pub id: WindowId,
 }
 
+/// The parts of a window, pending construction, that are dependent on top level app state.
+/// This includes the boxed root widget, as well as other window properties such as the title.
+pub struct PendingWindow<T> {
+    pub(crate) root: Box<dyn Widget<T>>,
+    pub(crate) title: LabelText<T>,
+    pub(crate) menu: Option<MenuDesc<T>>,
+}
+
+impl<T: Data> PendingWindow<T> {
+    /// Create a pending window from a boxed widget.
+    pub fn new_from_boxed(root: Box<dyn Widget<T>>) -> PendingWindow<T> {
```

Comment:
this jumps out at me; I'm sure there's a rationale it just isn't immediately clear. Is there a reason for this to be public API? Does it make sense to only have `fn new`, and make that do the boxing? Are there places where the incoming widget is already boxed, and we want to avoid boxing in that case?

Reply:
Got rid of it. Not sure if that is good or bad. It was like this just to preserve existing functionality as far as I remember.